### ISSUE TITLE
calculate number of interactions per datasetID x study

### ISSUE THREAD

**Issue by lizzieinvancouver** [lizzieinvancouver/ospree#249]
We need to count up the number of interactions across:
- photoperiod_day
- force
- chill
- chill days
- field.sample date
On both the BB data for OSPREE and the full OSPREE database (minus woody species). Please take a stab at it!

Since you should always start on a simple example I provide one here for photo x force:

```
> nointxn <- data.frame(datasetID=rep("bob12", 8), study=c(rep("exp2", 4), rep("exp2", 4)), photo=c(8, 8, 8, 8, 8, 8, 12, 12), force=c(15, 15, 18, 18, 15, 15, 15, 15))
> wintxn <- data.frame(datasetID=rep("bob14", 8), study=rep("exp1", 8), photo=c(8, 8, 8, 8, 12, 12, 12, 12), force=c(20, 20, 25, 25, 20, 20, 25, 25))
> testdat <- rbind(nointxn, wintxn)
> testdat
   datasetID study photo force
1      bob12  exp2     8    15
2      bob12  exp2     8    15
3      bob12  exp2     8    18
4      bob12  exp2     8    18
5      bob12  exp2     8    15
6      bob12  exp2     8    15
7      bob12  exp2    12    15
8      bob12  exp2    12    15
9      bob14  exp1     8    20
10     bob14  exp1     8    20
11     bob14  exp1     8    25
12     bob14  exp1     8    25
13     bob14  exp1    12    20
14     bob14  exp1    12    20
15     bob14  exp1    12    25
16     bob14  exp1    12    25
```

All welcome to participate in this challenge! Especially @dbuona @cchambe12 @AileneKane @MoralesCastilla since this will help the limiting cues paper. That said, it is not mandatory for anyone.

I put this code and the OSPREE file to use in analyses/limitingcues/countintxns.

**Comment by lizzieinvancouver** [comment]
@lizzieinvancouver Needs to follow up on this!

**Comment by cchambe12** [comment]
I can mostly get this to work but I'm coming across an issue.... some datasetIDs are doing multiple things but they aren't broken up by study so I'm not picking up on them in my new code. I've tried making the function run over each `figure.table..if.applicable.` but it makes the function super slow. Does anyone have any other suggestions on how to parse these datasetIDs apart other than study? Maybe species? respvar doesn't work either unfortunately. The study I'm thinking about specifically is basler14.

**Comment by cchambe12** [comment]
Just kidding, I figured out a work around! Works with test data and hopefully with the real data!

**Comment by lizzieinvancouver** [comment]
@cchambe12 Your code helped me figure out an error in my code (thank you)! I need to look more closely but I think our results almost match now.

**Comment by AileneKane** [comment]
Is this now done @lizzieinvancouver ?

**Comment by lizzieinvancouver** [comment]
@AileneKane Yes, I just need to set up a way to make sure the code uses what *you* use for the OSPREE manuscript ... I think you say there are 74 studies or such in the main text, can you tell me how to get that number (I assume it is a certain set of flags in models_stan or such) and I will update my code and put the numbers in the main ms file ... thanks!

**Comment by AileneKane** [comment]
here is a code snippet used to get the numbers in the manuscript (I may not have fully updated all parts of the manuscript but will shortly!): 
https://github.com/lizzieinvancouver/ospree/blob/master/notes/snippet_for_ns.R

**Comment by lizzieinvancouver** [comment]
I think this is done! Or close, @AileneKane I made two files you can use Sweave to ref, they are both in bb_anlaysis/output:
- bbstan_allsppmodel_countinxns.csv
- bbstan_mainmodel_countinxns.csv
I think they are straightforward to ref the numbers you need, but let me know if you have any questions! Otherwise, once they are ref-ed in the supp I can check and close this.

**Comment by lizzieinvancouver** [comment]
@AileneKane I have updated these files:

- bbstan_allsppmodel_countinxns.csv
- bbstan_mainmodel_countinxns.csv

To include everything you might need to ref. Can you read them into the main text and supp and Sexpr them (using the names in treat1 column ... that way we can update as needed ... but the numbers are in n column)? I think they are straightforward but here are some details (considering bbstan_allsppmodel_countinxns.csv ... but we should give this across allspp and main model I think):

Off the 66 (use total-datasetIDstudies) total studies in our all spp dataset ...
11 (total-exp-chill) manipulated chilling directly, while 24 (total-fielddate-chill) used multiple field sample dates (cite Weinberger)

Of the 23 (total-varied-forcing) studies that varied forcing, 14 (total-wsometreats-constant-forcing) included at least one treatment with constant temperatures across 24 hours, while 18 (total-variedwsometreats-varydaynight-forcing) at least one treatment with thermoperiodicity. 

[_Note  to us that studies often had a constant daily temp and a temp with varying daynight temps so these two numbers can add up to greater than 23 studies .... in contrast a single study could ONLY be assigned to exp chill or field chill.... not both, even though I think one had both (but it ended up in exp chill)._]

And all the interactions are listed too (total-exp-chill-fpintxns for example is all experiments that varied chill experimentally *and* also varied forcing or photoperiod). 

Let me know if this makes sense!

**Comment by AileneKane** [comment]
This makes sense. I have pushed a new version of the ms with these added. I will add the thermoperiodicity ones to the supp soon!

**Comment by lizzieinvancouver** [comment]
@cchambe12 I know I asked you this before so apologies for asking again, but can you remind me again why some datasetID x study show up multiple times in analyses/output/studytype_table.csv? For example caffarra11a  exp;  falusi97  exp1 ... Thank you again!

**Comment by cchambe12** [comment]
@lizzieinvancouver No worries! It's because there are multiple species in those studies and sometimes different species received a different number of treatments. If you check out analyses/output/studytype_withBB.csv it has a better breakdown. Should I make a note of this somewhere do you think? Or maybe add in species to the studytype_table.csv?

**Comment by lizzieinvancouver** [comment]
@cchambe12 Okay! That's what I thought. Yes! Could we add species to studytype_table.csv? *And* I believe studytype_withBB.csv is only budburst studies, could we have a version of it for all woody species OSPREE studies (on bbculdesac)? Then I could use that I believe.

**Comment by cchambe12** [comment]
@lizzieinvancouver sure thing! So should I use ospree_clean_withchill.csv for this or just the ospree.csv?

**Comment by lizzieinvancouver** [comment]
@cchambe12 Thanks! Can you use this:
```
dat <- read.csv("output/ospree_clean.csv", header = TRUE)
dat <- dat[dat$woody=="yes",]
```
And, can you help me understand why it shows caffarra11a  exp1 and exp3 as having 6 forcing treatments (or am I mis-reading this?) ... I checked the paper and am pretty sure that only exp2 varies forcing (see Table 1)? And similarly I think photo only varies in exp3 ...

**Comment by cchambe12** [comment]
@lizzieinvancouver great all set! And actually adding in the new columns fixed the issues you mentioned! I'm going to update these files on 'master' as well.

**Comment by lizzieinvancouver** [comment]
@cchambe12 Thank you! If you want (and you committed these changes along with no major others) you can try https://git-scm.com/docs/git-cherry-pick to just push the commit over to master.

**Comment by cchambe12** [comment]
gomory15 multiple forcing in @cchambe12 code but not in @lizzieinvancouver - double check paper and double check code @cchambe12

**Comment by lizzieinvancouver** [comment]
@cchambe12 

Let's go back, let's go back
Let's go way on way back when ....

Can you check your code to build output/studytype_table.csv (bbculdesac) and see gomory15 shows up with two forcing treatments? I think it's the main discrepancy between that table and my countinxns.R code. Thanks! And zero rush.

**Comment by cchambe12** [comment]
@lizzieinvancouver yes I see that now! For my code, I used ospree_clean.csv (which I believe you used too...) and it looks like the seedlings growing at the lower elevation experienced 9.2C forcing and the seedlings at the higher elevation experienced 4.9C. Based on the data and the paper, it looks like both species were divided evenly between the two growing elevations.

**Comment by lizzieinvancouver** [comment]
@cchambe12 Thank you! I figured it out from this (finally)! My code strips out any data that is not taken two weeks apart in field sampling date -- since this paper co-varied the forcing treatment with the sampling date we lose the earlier (1 Apr) sampling date, which is also the 4.9C treatment and just retain the 10 Apr sampling date. I think that's okay, but I will make a note. 

Thank you again! It's amazing how much work this is, and how nice it is lately to find that the data and code are more often good to go than not.

**Comment by lizzieinvancouver** [comment]
Some remaining tasks ...
* Check that these should all truly be constant photoperiod (all exp 1: "cronje03"   "devries82"  "karlsson03") my notes in studydesignplots.R says: # looked up the constant paper (karlsson03) and it's room ambient daylength I think
* review the checkchillist (in countintxns.R)
* see work on NAs (ask lab to check?) in studydesignplots.R

**Comment by lizzieinvancouver** [comment]
In ospree_clean_withchill_BB.csv ....
* karlsson03 -- photo_type is exp
* cronje03-- photo_type is exp
* devries82 is excluded

also ...

```
goober <- read.csv("~/Documents/git/projects/treegarden/budreview/ospree/analyses/output/ospree_clean_withchill_BB.csv")
try <- goober[which(paste(goober$datasetID, goober$study, sep="_") %in% checkchillist),]
boop <- subset(try, select=c("datasetID", "study", "chill_type"))
boop1 <- boop[!duplicated(boop),]
> boop1
        datasetID study chill_type
1     chavarria09  exp1        exp
27       falusi90  exp1        exp
39       falusi90  exp1       <NA>
421       jones12  exp2    bothest
423       jones12  exp1    bothest
442    junttila12  exp1        exp
466          li05  exp1        exp
497       rinne94  exp1        exp
500       rinne97  exp1        exp
1060        boyer  exp1    bothest
1130  caffarra11a  exp3        exp
1191  caffarra11b  exp3    bothest
1222   campbell75  exp1    bothest
1238   campbell75  exp3    bothest
1318    cannell83  exp1       <NA>
1319    cannell83  exp2       <NA>
1560     cronje03  exp1        exp
1988    granhus09  exp1       <NA>
2015  guerriero90  exp2    bothest
2028      Heide03  exp1        exp
3378      heide12  exp1    bothest
3544   karlsson03  exp1     fldest
3574       lamb37  exp1        exp
3645     laube14a  exp1     fldest
4000     myking95  exp1        exp
4315 nienstaedt66  exp1    bothest
5279   skuterud94  exp1    bothest
5299   skuterud94  exp1    bothrep
5405    sogaard08  exp2        exp
5427    sogaard08 exp1b        exp
5435    sogaard08 exp3b        exp
6283   sonsteby14  exp2    bothest
6287   sonsteby14  exp3    bothest
6333   sonsteby14  exp4    bothest
6413   sonsteby14  exp1    bothest
6525     spiers74  exp1    bothest
6567     swartz81  exp1    bothest
6571   thielges75  exp1        exp
6618       webb78  exp1    bothest
6662    worrall67  exp2       <NA>
6668    worrall67 exp 3        exp
6671    worrall67 exp 4        exp
>
```

**Comment by lizzieinvancouver** [comment]
and here's the NAs from studydesignplots.R 

```
> unique(checkforcena$datasetID)
[1] "ashby62"   "gansert02" "hawkins12" "ruesink98"
> unique(checkphotona$datasetID)
[1] "falusi96"    "gianfagna85" "manson91"    "nishimoto95"
> unique(checkchillna$datasetID)
 [1] "ashby62"        "basler12"       "basler14"       "biasi12"        "boyer"          "caffarra11a"   
 [7] "caffarra11b"    "calme94"        "cannell83"      "charrier11"     "cook00b"        "cook05"        
[13] "dantec14"       "devries82"      "falusi03"       "falusi96"       "falusi97"       "fu13"          
[19] "gansert02"      "ghelardini10"   "gianfagna85"    "gomory15"       "guak98"         "guerriero90"   
[25] "gunderson12"    "hawerroth13"    "hawkins12"      "heide05"        "heide08"        "heide11"       
[31] "heide12"        "heide93"        "heide93a"       "howe95"         "karlsson03"     "laube14a"      
[37] "laube14b"       "li05"           "manson91"       "morin10"        "myking98"       "nishimoto95"   
[43] "okie11"         "pagter15"       "partanen05"     "partanen98"     "pettersen71"    "ramos99"       
[49] "Sanz-Perez09"   "sanzperez10"    "schnabel87"     "sogaard08"      "sonsteby13"     "sonsteby14"    
[55] "spiers74"       "thielges75"     "viheraaarnio06" "webb78"         "worrall67"      "yazdaniha64"   
[61] "zohner16"  
```

**Comment by lizzieinvancouver** [comment]
```
> unique(checkchillna$datasetID)[which(!unique(checkchillna$datasetID) %in% checkchillna.helper$datasetID)]
 [1] "basler12"       "caffarra11a"    "cannell83"      "charrier11"     "cook05"         "devries82"     
 [7] "falusi96"       "hawerroth13"    "heide05"        "heide11"        "howe95"         "karlsson03"    
[13] "laube14a"       "laube14b"       "li05"           "morin10"        "pettersen71"    "sogaard08"     
[19] "sonsteby13"     "thielges75"     "viheraaarnio06" "worrall67" 
```

**Comment by lizzieinvancouver** [comment]
### constant photoperiod papers ...
- **cronje03** -- were take from field: "wrapped in moist paper before cold storage in plastic bags in an uprightposition at 4-7°C.... transferred to growth chamber maintained at a constant 25°C and illumination of 215 (imolm^s1 photo- synthetically active radiation, to force budburst" ... so this should be 24 h I think
- **devries82** -- scant methods but assuming 24 h seems reasonable
- **karlsson03** -- I don't think this paper says the photoperiod, "At monthly intervals from 15 January to 15 May, 2001, branches were taken from five (Kevo) or six (Abisko) trees and brought into the laboratory. The branches where then put into tap water and one branch from each tree were placed in three rooms with different temperatures, viz., 23–24, 16–18 and 11–13°C."

Take home: change cronje03 and devries82 to 24, karlsson03 should be unknown I think.

**Comment by lizzieinvancouver** [comment]
### Forcing NA
- ashby62 -- forcing not given as best I can tell
- gansert02 -- this is ambient I think, but we listed as none in BB data so leave as none
- hawkins12 -- common garden, okay to drop
- ruesink98 -- some (8-10 rows) are empty, we but we should have most of this study having forcetemp


### Photoperiod NA
- falusi96 -- mix of ambient (37 rows, all exp1), none (4 rows, exp 2) and experiment (8 rows, exp3), should check what's dropping out
- gianfagna85 -- not given!
- manson91 -- field warming study (on Kiwi) so ambient
- nishimoto95  -- not given!

Chilling NA (and checkchilllist)

For the checkchillist and the chill NA I think I should write up code that merges in info from the BB dataset in R and then work on the remaining issues.

**Comment by lizzieinvancouver** [comment]
Across checkchilllist and checkchillna I have <NA> for ...

ashby62 (exp 1)
caffarra11a (exp 1)
cannell83 (exp 1-2)
charrier11 (exp 1-2)
falusi90 exp1 (only a few rows I think) 
cook05 (exp 1)
gansert02 (exp 1-2)
granhus09 (exp 1)
heide05 (exp 1-2)
heide11 (exp 1-3)
howe95 (exp 1)
nishimoto95 (exp 1)
pettersen71 (exp 1-2)
sonsteby13 (exp 1 - 2)
worrall67 exp 1, 2, 5

**Comment by lizzieinvancouver** [comment]
Picking up where I left off a month ago (yikes)...

- gansert02 (exp 1-2) -- this an observational study (on Mt Fuji) of bud phenology -- it's across altitudes but it's ALL in the field so **it's not really an experiment.** 
- granhus09 (exp 1) -- these are the chilling interruption folks ("Chilling treatment
at 0.7 ± 0.7  ̊C interrupted by mild spells of 14 days duration at a constant temperature of 8 or 12  ̊C")
- heide05 (exp 1-2) -- the treatments sound like forcing ones, but it's not very clear to me what happened to the plants before the experiments started 
- heide11 (exp 1-3) -- similar to heide05 but it sounds like they were seeds or 'plantlets' (from a nursery) before the experiment started, so I would drop them as unclear (probably no chilling)
- howe95 (exp 1) -- sounds like rooted cuttings with no chill (as in, they got them 'rooted' and then started treatments)
- nishimoto95 (exp 1) -- for experiment with cuttings, they say they were field chilled, for plants they were at warm temps (28/15 day/night) in a greenhouse
- pettersen71 (exp 1-2) -- another 'young plant' study, seems like it's similar to heide work
- sonsteby13 (exp 1 - 2) -- "Vigorously growing, 1-year-old shoots were selected and labelled, different shoots being used each year" and then they cut them in the summer to determine flowering stafe
- worrall67 exp 1, 2, 5 -- 1 is an observational study of leafout in a greenhouse; 2 is cuttings from the field (field chilled until 10 Feb); and 5 is cuttings from the field taken on 15 Feb and planted into greenhouse

**Comment by lizzieinvancouver** [comment]
Based on my Oct 8 notes I changed manson91 to ambient photo in studydesignplots. R.

What's left? While for experimental chill (drawing from the above and studydesignplots.R code comparing NAs and where we say otherwise in the BB chill_type column) we should check if we can assign an average value or 'other' or 'variable' to ...
- cannell83 (exp 1-2)
- falusi96 (exp 2-3)
- sogaard08 (exp1a, exp3a, exp3b)
- viheraaarnio06 (exp 1-3)
- caffarra11b  exp2 (bothest)
- falusi97  exp2  (bothest)
-  biasi12  exp1 (bothrep) 
- cook00b  exp1 (bothrep) 
- heide93  exp3 (bothrep)

But first I should figure out how field chill shows up for those and make a plot ...
